src/event_sources/ble_esp32.h

```diff
@@ -95,13 +98,99 @@ class GATTResource : public BLEResource {
   }
 
  private:
-  uint16 _handle;
+  uint16 _handle = kInvalidHandle;
   Mutex* _mutex;
   uint32 _result = 0;
   uint32 _error = 0;
   struct os_mbuf* _mbuf = null;
 };
 
+
+class BLEServerServiceResource;
+typedef LinkedList<BLEServerServiceResource> BLEServerServiceList;
+
+class BLEServerCharacteristicResource;
+typedef LinkedList<BLEServerCharacteristicResource> BLEServerCharacteristicList;
+
+class BLEServerCharacteristicResource: public Resource, public BLEServerCharacteristicList::Element{
+ public:
+  TAG(BLEServerCharacteristicResource);
+  BLEServerCharacteristicResource(ResourceGroup* resource_group, BLEServerServiceResource* service,
```

Comment:
This appears leaky. What happens when the resource group is torn down and deletes this resource? Do you just leak the mbufs, etc.?